### PR TITLE
Cleanup javascript

### PR DESCRIPTION
**Pull request by Katrienvh**

### CODE REVIEW

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const locale = new Locale()
 
 export class Form {
   constructor (form) {
```
Comment (StijnVrolijk) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: What's this form in the constructor? I don't see it getting used anywhere

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
 export class Link {
+  constructor () {
+    this.initLink()
+  }
```
Comment (StijnVrolijk) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js]: Missing whitespace under this bracket

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js (hunk 1 above)
Comment (jonasdekeukelaere) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: Used to be used in SearchForm I think and for datefields, but I think this has been removed in the ES6 update, so might be obsolete now.

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ const locale = new Locale()
...
+    this.initForm()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: Can't the initForm code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js [hunk 4]
```diff
@@ -1,4 +1,7 @@
 export class Link {
+  constructor () {
+    this.initLink()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js]: Can't the initLink code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/LoadingBar.js [hunk 5]
```diff
@@ -1,4 +1,8 @@
 export class LoadingBar {
+  constructor () {
+    this.initEventListeners()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/LoadingBar.js]: Can't the initEventListeners code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Navbar.js [hunk 6]
```diff
@@ -1,4 +1,8 @@
 export class Navbar {
+  constructor () {
+    this.initNavbar()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Navbar.js]: Can't the initNavbar code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js [hunk 7]
```diff
@@ -1,4 +1,8 @@
 export class Scrolling {
+  constructor () {
+    this.initEventListeners()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js]: Can't the initEventListeners code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Searchbar.js [hunk 8]
```diff
@@ -1,4 +1,8 @@
 export class Searchbar {
+  constructor () {
+    this.initEventListeners()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Searchbar.js]: Can't the initEventListeners code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Table.js [hunk 9]
```diff
@@ -1,4 +1,8 @@
 export class Table {
+  constructor () {
+    this.initEventListeners
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Table.js]: Can't the initEventListeners code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js [hunk 10]
```diff
@@ -1,4 +1,9 @@
 export class Tabs {
+  constructor () {
+    this.initEventListeners()
```
Comment (jonasdekeukelaere) [inline src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js]: Can't the initEventListeners code not come in the constructor?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js (hunk 3 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js (hunk 4 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Link.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/LoadingBar.js (hunk 5 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/LoadingBar.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js (hunk 7 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Scrolling.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Searchbar.js (hunk 8 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Searchbar.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Table.js (hunk 9 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Table.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js (hunk 10 above)
Comment (mathiashelin) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Tabs.js]: Wouldn't that make the constructor a bit messy?

File: src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js (hunk 3 above)
Comment (jonasdekeukelaere) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: It's not like it's 10 lines of code, then I would start considering it, but in this case just 2.

Comment (StijnVrolijk) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: I agree with @jonasdekeukelaere but I wouldn't do it. If some code gets added later on that would mean refactoring the code _again_. Just calling these methods in the constructors is a lot better than it was before.

Comment (Katrienvh) [reply src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js]: i agree with @StijnVrolijk

**Pull request merged**